pkg/kubelet/images/image_manager.go

```diff
@@ -101,25 +106,33 @@ func NewImageManager(
 
 // imagePullPrecheck inspects the pull policy and checks for image presence accordingly,
 // returning (imageRef, error msg, err) and logging any errors.
-func (m *imageManager) imagePullPrecheck(ctx context.Context, objRef *v1.ObjectReference, logPrefix string, pullPolicy v1.PullPolicy, spec *kubecontainer.ImageSpec, requestedImage string) (imageRef string, msg string, err error) {
+func (m *imageManager) imagePullPrecheck(
+	ctx context.Context,
+	objRef *v1.ObjectReference,
+	logPrefix string,
+	pullPolicy v1.PullPolicy,
+	spec *kubecontainer.ImageSpec,
+	requestedImage string,
+) (imageRef string, imageFound *bool, msg string, err error) {
 	switch pullPolicy {
 	case v1.PullAlways:
-		return "", msg, nil
+		return "", nil, msg, nil
 	case v1.PullIfNotPresent, v1.PullNever:
 		imageRef, err = m.imageService.GetImageRef(ctx, *spec)
 		if err != nil {
 			msg = fmt.Sprintf("Failed to inspect image %q: %v", imageRef, err)
 			m.logIt(objRef, v1.EventTypeWarning, events.FailedToInspectImage, logPrefix, msg, klog.Warning)
-			return "", msg, ErrImageInspect
+			return "", nil, msg, ErrImageInspect
 		}
 	}
 
-	if len(imageRef) == 0 && pullPolicy == v1.PullNever {
+	imageFound = ptr.To(len(imageRef) > 0)
+	if !*imageFound && pullPolicy == v1.PullNever {
 		msg, err = m.imageNotPresentOnNeverPolicyError(logPrefix, objRef, requestedImage)
-		return "", msg, err
+		return "", ptr.To(false), msg, err
```

Comment:
why not return imageFound here? it should be pointer to false right?

Reply:
it's a code paranoia thing, I can change it to imageFound if you prefer